### PR TITLE
Use translations from this repository instead of db for categories 

### LINKED ISSUE #2923: Move category translations out of the database in our repos
We should move categories out of the database and in a python module in our repos, and then translate them using regular gettext calls. This will help performance and make it possible for translators to translate categories again.

This has already been done once before, in Marketplace:
https://github.com/mozilla/zamboni/commit/76927e730e7a8ac0d08a32dfeed4f43fbe6828c1
https://github.com/mozilla/zamboni/commit/0214acab25fa1d078c4f43373bf8805293534078

One thing to be careful about is that AMO has different categories for different applications. Maybe we can unify all that though.

### PR DESCRIPTION
**Pull request by diox**
Fix #2923

### CODE REVIEW

**Comment by diox** [comment]
The real changes, drowned in a sea of fixed tests, are in the `Category` model: https://github.com/mozilla/addons-server/pull/3210/files#diff-d5d4f5e227c5a588706b9e9f6879197eR1841

File: src/olympia/addons/models.py [hunk 1]
```diff
@@ -1857,6 +1861,16 @@ class Meta:
...
         verbose_name_plural = 'Categories'
 
+    @property
+    def name(self):
+        try:
+            value = categories[self.application][self.type][self.slug].name
+        except KeyError:
+            # If we can't find the category in the constants dict, fall back
+            # to the db field.
+            value = self.db_name
```
Comment (muffinresearch) [inline src/olympia/addons/models.py]: Should we log the fallback cases?

**Comment by muffinresearch** [comment]
Unrelated to this change but is the landfill code still needed?

**Comment by muffinresearch** [comment]
lgtm r+wc

It'll definitely be worth adding some notes so QA know how to validate this and what to look out for and maybe add the `needs more qa` label too?

**Comment by diox** [comment]
The landfill code is still used to populate local dev environments ; I should be able to simplify the categories generation even further, but that's not my priority at the moment.

File: src/olympia/addons/models.py (hunk 1 above)
Comment (diox) [inline src/olympia/addons/models.py]: Yep, good idea. It *should* not be used much, logging will tell us.

**Pull request merged**